internal/auth0/email/resource.go

```diff
@@ -0,0 +1,248 @@
+								"possible because many providers have limitations on this port. Used only for SMTP.",
+						},
+						"smtp_user": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							ValidateFunc: validation.StringIsNotEmpty,
+							Description:  "SMTP username. Used only for SMTP.",
+						},
+						"smtp_pass": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Sensitive:    true,
+							ValidateFunc: validation.StringIsNotEmpty,
+							Description:  "SMTP password. Used only for SMTP.",
+						},
+					},
+				},
+			},
+			"settings": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Computed:    true,
+				Description: "Specific email provider settings.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"message": {
+							Type:        schema.TypeList,
+							MaxItems:    1,
+							Optional:    true,
+							Description: "Message settings for the `mandrill` or `ses` email provider.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"view_content_link": {
+										Type:     schema.TypeBool,
+										Optional: true,
+										Description: "Setting for the `mandrill` email provider. " +
+											"Set to `true` to see the content of individual emails sent to users.",
+									},
+									"configuration_set_name": {
+										Type:     schema.TypeString,
+										Optional: true,
+										Description: "Setting for the `ses` email provider. " +
+											"The name of the configuration set to apply to the sent emails.",
+									},
+								},
+							},
+						},
+						"headers": {
+							Type:        schema.TypeList,
+							MaxItems:    1,
+							Optional:    true,
+							Description: "Headers settings for the `smtp` email provider.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"x_mc_view_content_link": {
+										Type:     schema.TypeString,
+										Optional: true,
+										ValidateFunc: validation.StringInSlice(
+											[]string{"true", "false"},
+											false,
+										),
+										Description: "Disable or enable the default View Content Link " +
+											"for sensitive emails.",
+									},
+									"x_ses_configuration_set": {
+										Type:        schema.TypeString,
+										Optional:    true,
+										Description: "SES Configuration set to include when sending emails.",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func createEmail(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	api := m.(*management.Management)
+
+	d.SetId(resource.UniqueId())
+
+	if emailProviderIsConfigured(api) {
+		return updateEmail(ctx, d, m)
+	}
+
+	email := expandEmailProvider(d.GetRawConfig())
+	if err := api.EmailProvider.Create(email); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return readEmail(ctx, d, m)
```

Comment:
Is it necessary to call `readEmail()` here, given that `email` will be hydrated after the `Create()` call?

Reply:
You're right, this is not necessary at all. However it is a pattern that we inherited when taking over the provider. All create funcs will end up with a read in order to avoid reimplementing a couple of `data.Set` calls. I had this in the back of my mind to eventually fix so we can cut a few of those unnecessary API calls. If time permits I might be able to work on this next week.